test/lightning/credentials/keychain_credential_test.exs

```diff
@@ -411,4 +411,72 @@ defmodule Lightning.Credentials.KeychainCredentialTest do
       assert loaded_keychain.default_credential.id == credential.id
     end
   end
+
+  describe "external_id uniqueness validation" do
```

Comment:
Nice test documenting the failure mode for legacy data! But this makes me think, users in prod might already have duplicates in their database, right? They'd hit this `Ecto.MultipleResultsError` at runtime. Maybe worth a follow-up issue to add a migration or mix task that detects and reports existing duplicates so admins know to clean them up?

Reply:
yeah, good call. i was planning to update known instances manually but we should include a migration here. ill do so